### PR TITLE
Make the Relay block teleports

### PR DESCRIPTION
**Pull request by FlameArrow57**
[GAME MODES][BALANCE][CLEANLINESS]

## About the PR 
This PR just makes the relay a teleportation jammer.

Slightly reworks teleport jamming code to do this

## Why's this needed? 
Prevents late game bomb teleports, etc.

Also in theme of Flock, Flock lore has the Flock hating teleportation

### CODE REVIEW

**Comment by FlameArrow57** [comment]
Saw that do_teleport() doesn't check if teleportation is blocked at the source of the teleport, relevant to hand teles. Could change that if wanted

Alternative could be making all Flock floortiles have a chance to or completely block teleports

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]
Perfect candidate for just making into a component rather than sticking another var on atom

**Review by TobleroneSwordfish — APPROVED** [review]
that works 👍

File: _std/macros/atom_properties.dm [hunk 1]
```diff
@@ -255,6 +255,7 @@ To remove:
 //-------------------- ATOM PROPS -----------------------
 #define PROP_ATOM_NEVER_DENSE(x) x("neverdense", APPLY_ATOM_PROPERTY_SIMPLE, REMOVE_ATOM_PROPERTY_SIMPLE)
 #define PROP_ATOM_NO_ICON_UPDATES(x) x("no_icon_updates", APPLY_ATOM_PROPERTY_SIMPLE, REMOVE_ATOM_PROPERTY_SIMPLE)
+#define PROP_ATOM_TELEPORT_JAMMER(x) x("teleport_jammer", APPLY_ATOM_PROPERTY_SUM, REMOVE_ATOM_PROPERTY_SUM)
```
Comment (Tarmunora) [inline _std/macros/atom_properties.dm:258]: Property should automatically add/remove atoms from the teleblocker tracking cat when added/removed (see `PROP_UPDATE_INVISIBILITY` etc)

**Review by Tarmunora — CHANGES_REQUESTED** [review]

File: _std/macros/atom_properties.dm [hunk 2]
```diff
@@ -168,6 +168,15 @@ To remove:
 	_sightprocess?.Process(); \
 } while(0)
 
+#define PROP_UPDATE_TELEBLOCK_CAT(target, prop, old_val) do { \
+	var/is_teleblocking = GET_ATOM_PROPERTY_RAW(target, prop); \
+	if (isnull(old_val) || !isnull(is_teleblocking)) { \
```
Comment (Tarmunora) [inline _std/macros/atom_properties.dm]: not `&&`?

Comment (FlameArrow57) [reply _std/macros/atom_properties.dm]: Was fixed in recent commit

**Review by Tarmunora — APPROVED** [review]

**Pull request merged**